### PR TITLE
Cleanup obsolete `PBIS` support (removed in XS8) from extauth-hook-AD.py

### PR DESCRIPTION
**Pull request by bernhardkaindl**
To: @liulinC (Author of the removed lines)

XS8 removed PBIS Active Directory support (was replaced by Winbind).

- I assume, that PBIS code could be removed from Xapi therefore.

Testing:
- The unit test `python3/plugins/test_extauth_hook_AD.py` provides code coverage for 100% of the changed lines.
- The current `diff-cover` configuration that runs in `pre-commit` and GitHub CI ensures that indeed 100% of all changed lines are covered by the test.

----

PS:

These OCaml sources could be removed too:
```py
ocaml/tests/test_extauth_plugin_ADpbis.ml
ocaml/xapi/extauth_plugin_ADpbis.ml
scripts/pbis-force-domain-leave
```

They are needed by those (PBIS is also referred to in those and would need changes too):
```py
ocaml/tests/test_extauth_plugin_ADwinbind.ml
ocaml/tests/suite_alcotest.ml
ocaml/xapi/extauth_plugin_ADwinbind.mli
ocaml/xapi/xapi_globs.ml
ocaml/xapi/extauth_ad.ml
ocaml/xapi/extauth_plugin_ADwinbind.ml
```

### CODE REVIEW

**Comment by bernhardkaindl** [comment]
Closing for opening again later when ready

**Comment by liulinC** [comment]
@bernhardkaindl  thank for helping on this.
Both the python file and ocaml file can be removed from master
- PBIS removed from XS8
- Yangtze use PBIS, branched out.
-
Better to do authentication XenRT verification.
Given this is target to feature/py3, so I approve this PR, 
We will run some authentication XenRT regression for this before merge to master.

File: python3/plugins/extauth-hook-AD.py [hunk 1]
```diff
@@ -265,16 +239,15 @@ def _add_upn(self, subject_rec):
 
     def _add_subject(self, subject_rec):
         try:
+            # ._format_item(name) would have formatted the name to be in the format
```
Comment (liulinC) [inline python3/plugins/extauth-hook-AD.py:242]: .format_item is removed, so no need to comment it.
BTW, BPIS-->PBIS

File: python3/plugins/extauth-hook-AD.py [hunk 2]
```diff
@@ -85,22 +84,14 @@ def __init__(self, path, session, args, ad_enabled=True, load_existing=True, fil
         self._session = session
         self._args = args
         self._lines = []
-        self._backend = self._get_ad_backend()
+        self._backend = ADBackend.BD_WINBIND
```
Comment (liulinC) [inline python3/plugins/extauth-hook-AD.py:87]: _backend is now only winbind, 
so we do not need `ADBackend` anymore.

File: python3/plugins/extauth-hook-AD.py (hunk 1 above)
Comment (bernhardkaindl) [reply python3/plugins/extauth-hook-AD.py]: Thanks, removed.

File: python3/plugins/extauth-hook-AD.py (hunk 2 above)
Comment (bernhardkaindl) [reply python3/plugins/extauth-hook-AD.py]: Thanks, removed.

**Comment by bernhardkaindl** [comment]
Thanks @liulinC, I applied the removal of the lines that you commented in your review:
https://github.com/xapi-project/xen-api/compare/ac02f32c720bb7e3bc9dfc28f003c91c9bd2ff2a..cc697ed48ecb105468be844606121d67e23630bd

**Review by psafont — APPROVED** [review]

**Comment by liulinC** [comment]
- This PR goes to feature branch, and needs to do XenRT authentication test (in CP-49162) before merge to master.
- This PR clean the plugin code, CP-50352 will handle the ocaml part.

**Review by liulinC — APPROVED** [review]

**Comment by liulinC** [comment]
@bernhardkaindl  Thanks for helping to removing the code, 
Could you please help to resolve the conflict to get it merged, thanks

**Comment by stephenchengCloud** [comment]
@bernhardkaindl
If you have other changes related to python3, could you hold on for a while, as we plan to do a regression test for python3 feature as recently there have been quite a lot PRs merged by Ashwin and you. Thanks.

**Comment by psafont** [comment]
The branch needs rebasing to resolve conflicts

File: python3/plugins/extauth-hook-AD.py [hunk 3]
```diff
@@ -260,15 +228,12 @@ def _add_upn(self, subject_rec):
     def _add_subject(self, subject_rec):
         try:
             sid = subject_rec['subject_identifier']
-            name = subject_rec["other_config"]["subject-name"]
-            formatted_name = self._format_item(name)
+            formatted_name = subject_rec["other_config"]["subject-name"]
```
Comment (liulinC) [inline python3/plugins/extauth-hook-AD.py:231]: formatted_name is no longer a proper name, but never mind.

**Pull request merged**